models/taxon.py

```diff
@@ -0,0 +1,41 @@
+import sqlalchemy as sql
+
+class Taxon():
+    @staticmethod
+    def search_by_name(db_conn, term, page=1, per_page=10):
+        term = term.lower().strip()
+        if len(term) <= 0:
+            return [], 0
+
+        term_pattern = '%' + '%'.join(term.split()) + '%'
+        first_word = term.split()[0]
+
+        query = """
+            SELECT
+                tax_id AS id,
+                tax_names AS text
+            FROM Taxa
+            WHERE LOWER(tax_names) LIKE :term_pattern
+            ORDER BY
+                LOCATE(:first_word, LOWER(tax_names)) ASC,
```

Comment:
A useful method that we can reuse -- the mysql [`LOCATE`](https://dev.mysql.com/doc/refman/8.4/en/string-functions.html#function_locate) function, returning the index of the match.